app/forms/validators.py

```diff
@@ -48,11 +49,17 @@ def __call__(
         form: FlaskForm,
         field: Union[DecimalFieldWithSeparator, IntegerFieldWithSeparator],
     ) -> None:
+        try:
+            if math.isnan(float(field.raw_data[0])):
```

Comment:
Why are we casting to a float?. Also error message should be `self.message` as this is in the context of a validator. Same for the other one.

Reply:
Just cause `nan` is a string here and it wouldn't evaluate as `True` if you try `math.isnan()` on a string. Added the self messages now. Thanks.

Reply:
You can remove this `try catch` approach and do this at the end:

```py
        if "e" in field.raw_data[0].lower() or math.isnan(field.data):
            raise validators.StopValidation(self.message)
```

If data is None this can error but this should never get called when data is None as validation like this only kicks in if we have a value. You can also use `field.data` will be the processed value.

Reply:
Also `math.isnan()` will error if input is not numerical value so we need this `try`?

Reply:
`field.data` isn't None when it is `NaN`, it will be `Decimal('NaN')`, same as you casting to float. So you should be able to use it as is.

Reply:
Correct me if I am wrong since I was wrong about `field.data` but do we not need to cast float before we do isnumeric() since `field.data` are strings? That's what i do now so it's:
```python
        if "e" in field.raw_data[0].lower() or math.isnan(float(field.data[0])):
            raise validators.StopValidation(self.message)

```

Reply:
Ok, thanks, the only edge case is when you have a whole number field and you enter decimal, it errors as the field data becomes `None` so I have to guard against it by checking if field data is not `None`.

Reply:
I see that you have done `if field.data`, we should update it to be explicit here, so `if field.data is not None and ...`

Reply:
Added, thanks!